multik-api/src/commonMain/kotlin/org/jetbrains/kotlinx/multik/ndarray/complex/_ComplexArrays.kt

```diff
@@ -1353,7 +1353,9 @@ public inline fun ComplexFloatArray.copyOfRange(fromIndex: Int, toIndex: Int): C
     val newLength = toIndex - fromIndex
     require(newLength >= 0) { "$fromIndex > $toIndex" }
     val ret = ComplexFloatArray(newLength)
-    System.arraycopy(this.getFlatArray(), fromIndex, ret.getFlatArray(), 0, min(this.size - fromIndex, newLength) * 2)
+    this.getFlatArray().copyInto(ret.getFlatArray(), 0,fromIndex, toIndex)
+    val endIndex = min(this.size - fromIndex, newLength) * 2
+    this.getFlatArray().copyInto(ret.getFlatArray(), fromIndex,0, endIndex)
```

Comment:
Here, apparently, I was mistaken, since I didn't multiply `fromIndex` by 2

Reply:
I'm confused. You think this function is correct, or incorrect? Can you explain how the memory is laidout for complex? I'm not even sure why I'm doing two separate copyIntos @devcrocod